### PR TITLE
Feature bit 263

### PR DESCRIPTION
**Pull request by fpelliccioni**
I can't merge it right now, but I will solve this problem in a few minutes.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -540,14 +548,14 @@ if (WITH_TESTS)
...
 
-  _add_tests(bitprim_blockchain_test 
-    block_entry_tests
-    block_pool_tests
-    branch_tests
-    transaction_entry_tests
-    validate_block_tests
-    validate_transaction_tests
-  )
+  # _add_tests(bitprim_blockchain_test
```
Comment (mdallago) [inline CMakeLists.txt]: why the tests are commented ?

File: conanfile.py [hunk 2]
```diff
@@ -133,10 +137,12 @@ def configure(self):
         self.options["*"].db = self.options.db
 
         self.options["*"].use_domain = self.options.use_domain
-        self.options["*"].mining = self.options.mining
+        # self.options["*"].mining = self.options.mining
+        # self.options["*"].mempool = self.options.mempool
```
Comment (mdallago) [inline conanfile.py:141]: this line is not required ?

File: include/bitcoin/blockchain/pools/transaction_organizer.hpp [hunk 3]
```diff
@@ -32,6 +32,12 @@
 #include <bitcoin/blockchain/settings.hpp>
 #include <bitcoin/blockchain/validate/validate_transaction.hpp>
 
+#if defined(BITPRIM_WITH_MEMPOOL)
+// #include <bitprim/mining/mempool_v1.hpp>
```
Comment (dario-ramos) [inline include/bitcoin/blockchain/pools/transaction_organizer.hpp]: Clean up old version

File: include/bitprim/mining/mempool_v1.hpp [hunk 4]
```diff
@@ -0,0 +1,1909 @@
...
+                    if (res.second) {
+                        fee += child.fee();
+                        size += child.size();
+                        sigops += child.sigops();
+                    }
+                }
+            }
+        }
+
+        // if (node.children_fees() != fee) {
```
Comment (dario-ramos) [inline include/bitprim/mining/mempool_v1.hpp]: Remove commented code, makes reading harder

File: include/bitprim/mining/mempool_v1.hpp [hunk 5]
```diff
@@ -0,0 +1,1909 @@
...
+        // }
+
+        BITCOIN_ASSERT(node.children_fees() == fee);
+        BITCOIN_ASSERT(node.children_size() == size);
+        BITCOIN_ASSERT(node.children_sigops() == sigops);
+    }
+
+
+    void check_invariant() const {
+        check_invariant_partial();
```
Comment (dario-ramos) [inline include/bitprim/mining/mempool_v1.hpp]: Which is the invariant?

File: include/bitprim/mining/mempool_v1.hpp [hunk 6]
```diff
@@ -0,0 +1,1909 @@
...
+                // auto res2 = std::is_sorted(candidate_transactions_.begin(), candidate_transactions_.end(), cmp);
+                // std::cout << res2;
+            // }
+
+            BOOST_ASSERT(res);
+        }        
+    }
+
+    void check_invariant_partial() const {
+
```
Comment (dario-ramos) [inline include/bitprim/mining/mempool_v1.hpp]: This method is too long, wouldn't it be more readable breaking it into inlined submethods?

File: include/bitprim/mining/mempool_v1.hpp [hunk 7]
```diff
@@ -0,0 +1,1909 @@
...
+                }
+            }            
+        }
+    }
+
+    template <typename I>
+    error::error_code_t remove(I f, I l, size_t non_coinbase_input_count = 0) {
+        // precondition: [f, l) is a valid non-empty range
+        //               there are no coinbase transactions in the range
+
```
Comment (dario-ramos) [inline include/bitprim/mining/mempool_v1.hpp:534]: This method is too long and complex, can it be simplified?

File: include/bitprim/mining/mempool_v1.hpp [hunk 8]
```diff
@@ -0,0 +1,1909 @@
...
+        auto const& node_b = all_transactions_[b];
+
+        auto const value_a = static_cast<double>(node_a.children_fees()) / node_a.children_size();
+        auto const value_b = static_cast<double>(node_b.children_fees()) / node_b.children_size();
+
+        return value_b < value_a;
+    }
+
+#if defined(BITPRIM_CURRENCY_BCH)
+    bool ctor_cmp(index_t a, index_t b) const {
```
Comment (dario-ramos) [inline include/bitprim/mining/mempool_v1.hpp:1299]: Can ctor and ltor be implemented in a separate file, or using strategy pattern?

File: include/bitprim/mining/mempool_v1.hpp [hunk 9]
```diff
@@ -0,0 +1,1909 @@
...
+                auto& parent = all_transactions_[pi];
+                if (parent.candidate_index() != null_index) {
+                    reindex_parent_for_removal(node, parent, pi);
+                }
+            }
+        }
+    }
+
+
+    void reindex_parent_from_insertion(mining::node const& node, mining::node& parent, index_t parent_index) {
```
Comment (dario-ramos) [inline include/bitprim/mining/mempool_v1.hpp:1851]: This method is very hard to follow

File: include/bitprim/mining/prioritizer.hpp [hunk 10]
```diff
@@ -0,0 +1,92 @@
+/**
```
Comment (dario-ramos) [inline include/bitprim/mining/prioritizer.hpp:1]: Isn't there a data structure like this one in std or boost?

File: test_new/mempool_tests.cpp [hunk 11]
```diff
@@ -0,0 +1,2763 @@
...
+    mp.check_invariant();
+}
+
+    
+TEST_CASE("[mempool] testnet case 0") {
+    mempool mp(20000);
+    
+    std::cout << "error::low_benefit_transaction: " << error::low_benefit_transaction << std::endl;
+
+    std::vector<std::string> olds {
```
Comment (dario-ramos) [inline test_new/mempool_tests.cpp]: Could all this data be moved to a separate file to avoid clutter?

File: include/bitprim/mining/mempool_v1.hpp (hunk 5 above)
Comment (dario-ramos) [reply include/bitprim/mining/mempool_v1.hpp]: Class methods

File: include/bitprim/mining/mempool_v1.hpp (hunk 6 above)
Comment (dario-ramos) [reply include/bitprim/mining/mempool_v1.hpp]: Marked with a TODO

File: include/bitprim/mining/mempool_v1.hpp (hunk 7 above)
Comment (dario-ramos) [reply include/bitprim/mining/mempool_v1.hpp:534]: Refactoring pending

File: include/bitprim/mining/mempool_v1.hpp (hunk 8 above)
Comment (dario-ramos) [reply include/bitprim/mining/mempool_v1.hpp:1299]: Solved in compile time, strategy makes no sense

File: include/bitprim/mining/mempool_v1.hpp (hunk 9 above)
Comment (dario-ramos) [reply include/bitprim/mining/mempool_v1.hpp:1851]: Refactoring pending

File: include/bitprim/mining/prioritizer.hpp (hunk 10 above)
Comment (dario-ramos) [reply include/bitprim/mining/prioritizer.hpp:1]: Ignore, there is no such structure

Comment (fpelliccioni) [reply include/bitprim/mining/prioritizer.hpp:1]: I did't find it, so I had to implement it.

File: test_new/mempool_tests.cpp (hunk 11 above)
Comment (dario-ramos) [reply test_new/mempool_tests.cpp]: Wrote TODO

**Review by dario-ramos — DISMISSED** [review]

**Review by mdallago — APPROVED** [review]
